source/fundamentals/aggregation.txt

```diff
@@ -7,51 +9,235 @@ Aggregation
+assembles the parts into a car.
+
+The assembly line resembles the **aggregation pipeline**, the assembly
+stations in the assembly line resemble the **aggregation stages**, the
+specialized tools represent the **expression operators**, and the
+finished product resembles the **aggregated result**. 
+
+Compare Operations
+------------------
+
+The following table lists the tasks you can perform with find and aggregation operations.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Find Operations
+     - Aggregation Operations
+
+   * - | Select *what* documents to return
+       | Select *which* fields to return
+       | Sort the results
+       | Limit the results
+       | Count the results
+     - | Select *what* documents to return
+       | Select *which* fields to return
+       | Sort the results
+       | Limit the results
+       | Count the results
+       | Rename fields
+       | Calculate fields
+       | Summarize data
+       | Group values
+
+Limitations
+-----------
+
+Aggregation operations have limitations. When performing agregation
+operations, keep the following in mind:
+
+- Returned documents must not violate the :manual:`BSON document size
+  limit </reference/limits/#BSON-Document-Size>` of 16 megabytes.
+- Pipeline stages have a memory limit of 100 megabytes by default. If
+  required, you may exceed this limit by using the `allowDiskUse
+  <{+api+}/mongo/options#AggregateOptions.SetAllowDiskUse>`__
+  method.
+- The :manual:`$graphLookup
+  </reference/operator/aggregation/graphLookup/>` stage
+  has a strict memory limit of 100 megabytes and ignores
+  ``allowDiskUse``.
+
+Examples
+--------
 
-     The :manual:`$graphLookup </operator/aggregation/graphLookup/>` stage has a strict memory limit of 100 megabytes
-     and will ignore ``allowDiskUse``.
+To run the examples in this section, load the sample data into the
+``tea.ratings`` collection with the following snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/aggregation.go
+   :start-after: begin insert docs
+   :end-before: end insert docs
+   :language: go
+   :dedent:
+
+Each document represents a person's rating for a particular tea, and the
+amount of visits they made to a local tea shop.
+
+Average Rating
+~~~~~~~~~~~~~~
+
+The following example calculates and displays the average rating and
+amount of ratings for each tea.
+
+The aggregation pipleline uses the ``$group`` stage to group the
+documents by the ``type`` field, calculates the average using the
+``$avg`` expression operator, and counts the number of documents using
+the ``$sum`` expression operator.
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: go
+
+      // create the stage
+      groupStage := bson.D{
+          {"$group", bson.D{
+              {"_id", "$type"},
+              {"average", bson.D{
+                  {"$avg", "$rating"},
+              }},
+              {"count", bson.D{
```

Comment:
Let's name this something more descriptive, like total. Then down below the output could be "total ratings"